hpi-swt2/lecture-portal
# Enable polls, feedback according to current lecture status

# Resolves #306

This PR enables/ disables interaction with feedback and polls according to the status of the lecture and active or running lectures don't have an Edit button on the index page any more. 

Acceptance Criteria:
- [x] Lecture interactivity features (questions, polls, feedback) only accept input while the lecture is active
- [x] The lecture states follow the behavior specified in the diagram (https://drive.google.com/file/d/164c58rieeQvvqHmFXsPfiEOPw1B6m8Ig/view?usp=sharing) (black parts)
    “Created” means: Students can’t do anything with the lecture. It appears in the lecture list of the course. The lecturer can view and edit the lecture.
    “Active” means: Students can join the lecture and engage in all interactivity offered by the lecture's configuration.
    “Archived” means: The lecture can be seen in the course's lecture list. The lecture can be viewed by the lecturer. No interactivity is allowed anymore.

Testing steps:
- change the status of a lecture of active/running: test if everything works according to the settings of the lecturer
- change the status to archived: Students should not be able to give feedback, Lecturers and students should only be able to see the results of a poll

---

Definition Of Done:
